exporter/SynthesisFusionAddin/src/Parser/SynthesisParser/Parser.py

```diff
@@ -249,3 +267,11 @@ def export(self) -> None:
         )
 
         logger.debug(debug_output.strip())
+
+
+def handle_err_top[T](err: Result[T]) -> None:
+    if err.is_err():
+        message, severity = err.unwrap_err()
+        if severity == ErrorSeverity.Fatal:
+            app = adsk.core.Application.get()
+            app.userInterface.messageBox(f"Fatal Error Encountered: {message}")
```

Comment:
Thoughts on making this some sort of function decorator? Feels a little verbose currently but I'm open to suggestions.

https://github.com/Autodesk/synthesis/blob/456a853d430ffc2246acfad51ec972d7809bc7d0/exporter/SynthesisFusionAddin/src/Parser/SynthesisParser/Parser.py#L82-L121